### PR TITLE
Excalicharts MVP

### PR DESCRIPTION
**Pull request by petehunt**
Copy some cells from excel, paste them into excalidraw, and get an editable bar chart.

In the future we may want to add something to the toolbar or more chart types. This is an MVP optimized for a small implementation size and for making Excalidraw an option for creating presentations.

Screen Shot 2020-06-05 at 11 33 45 AM

### CODE REVIEW

**Comment by dwelle** [comment]
Shouldn't the x/y axis be reversed. Let's say I have this data:

```
date	amount
2020-06-06	1
2020-06-07	23
2020-06-08	2
2020-06-09	4
2020-06-10	33
2020-06-11	42
2020-06-12	12
2020-06-13	34
```

it creates:

image

File: src/charts.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        strokeWidth: appState.currentItemStrokeWidth,
+        strokeStyle: appState.currentItemStrokeStyle,
+        roughness: appState.currentItemRoughness,
+        opacity: appState.currentItemOpacity,
+        fontSize: 16,
+        fontFamily: appState.currentItemFontFamily,
+        textAlign: "center",
+        width: BAR_WIDTH,
+      });
+      mutateElement(textElement, { angle: ANGLE });
```
Comment (dwelle) [inline src/charts.ts:232]:
```suggestion
        width: BAR_WIDTH,
        angle: ANGLE,
      });
```

File: src/charts.ts [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      return textElement;
```
Comment (dwelle) [inline src/charts.ts:233]:
```suggestion
      return newTextElement({
        text: label.length > 8 ? `${label.slice(0, 5)}...` : label,
        x: x + labelX,
        y: y + labelY,
        strokeColor: appState.currentItemStrokeColor,
        backgroundColor: appState.currentItemBackgroundColor,
        fillStyle: appState.currentItemFillStyle,
        strokeWidth: appState.currentItemStrokeWidth,
        strokeStyle: appState.currentItemStrokeStyle,
        roughness: appState.currentItemRoughness,
        opacity: appState.currentItemOpacity,
        fontSize: 16,
        fontFamily: appState.currentItemFontFamily,
        textAlign: "center",
        width: BAR_WIDTH,
        angle: ANGLE,
      });
```

Comment (dwelle) [reply src/charts.ts:233]: This just puts the `angle` into constructor.

**Comment by lipis** [comment]
@dwelle I think in that case they are both treated as numbers

**Comment by dwelle** [comment]
> @dwelle I think in that case they are both treated as numbers

But it should use 1st columns for X axis.

**Comment by petehunt** [comment]
ah that bug came from a last minute decision to remove a specific list of currency symbols. i added it back in -- it's just a kind of random selection of currency symbols I threw in there. is there a more official list we should use?

regardless, it might be ok to roll with this for now and fix it in a follow up since it's not super discoverable in this form anyway. alternatively, we could remove support for currencies as numbers.

**Comment by dwelle** [comment]
> it's just a kind of random selection of currency symbols I threw in there. is there a more official list we should use?

No idea, but it also won't be as simple as that because not every currency puts the symbol *before* the number.

But like you said, let's not overengineer it right now.

**Comment by vjeux** [comment]
In France you put the euro symbol -after- the number for example :)

**Comment by vjeux** [comment]
What I would do is to try and parse the number and ignore everything else. Also in French you use comma for decimals rather than dots.

**Pull request merged**